account_payment_partner/models/account_move.py

```diff
@@ -65,6 +66,14 @@ def _compute_payment_mode(self):
                     move.payment_mode_id = move.with_context(
                         force_company=move.company_id.id
                     ).partner_id.customer_payment_mode_id
+                elif move.type == "out_refund":
+                    move.payment_mode_id = move.with_context(
+                        force_company=move.company_id.id
+                    ).partner_id.customer_payment_mode_id.refund_payment_mode_id
```

Comment:
Another option can be to use `reversed_entry_id` field to evaluate if get payment mode from it or from partner.